moff.py

```diff
@@ -1,313 +1,300 @@
+    for c_name in col_must_have:
+        if not (c_name in col_list):
+            # fail
+            return 1
+    # succes
+    return 0
+
+
+def run_apex(file_name, tol, h_rt_w, s_w, s_w_match, loc_raw, loc_output):
+    # OS detect
+    flag_windows = False
+    if _platform in ["linux", "linux2", 'darwin']:
+        flag_windows = False
+    elif _platform == "win32":
+        flag_windows = True
+
+    # flag_for matching
+    mbr_flag = 0
+    config = ConfigParser.RawConfigParser()
+    # get the  running path of moff
+    moff_path = os.path.dirname(sys.argv[0])
+
+    # it s always placed in same folder of moff.py
+
+    config.read(os.path.join(moff_path, 'moff_setting.properties'))
+
+    # case of moff_all more than one subfolderi
+    name = os.path.basename(file_name).split('.')[0]
+    if '_match' in name:
+        # in case of mbr , here i dont have evaluate the flag mbr
+        start = name.find('_match')
+        # extract the name of the file
+        name = name[0:start]
+
+    log = logging.getLogger('moFF apex module')
+    log.setLevel(logging.INFO)
+    if loc_output != '':
+        if not (os.path.isdir(loc_output)):
+            os.makedirs(loc_output)
+            log.info("created output folder: ", loc_output)
+
+        # outputname : name of the output
+        # it should be ok also in linux
+        outputname = os.path.join(loc_output, name + "_moff_result.txt")
+        fh = logging.FileHandler(os.path.join(loc_output, name + '__moff.log'), mode='w')
+    else:
+        outputname = name + "_moff_result.txt"
+        fh = logging.FileHandler(os.path.join(name + '__moff.log'), mode='w')
+
+    fh.setLevel(logging.INFO)
+    log.addHandler(fh)
+
+    if loc_raw is not None:
+        if flag_windows:
+            loc = os.path.join(loc_raw, name + '.RAW')
+        else:
+            # raw file name must have capitals letters :) this shloud be checked
+            loc = os.path.join(loc_raw, name.upper() + '.RAW')
+    else:
+        # that must be tested for the windows vers.
+        loc = os.path.join(loc_raw, name + '.RAW')
+
+    if os.path.isfile(loc):
+        log.info('raw file exist')
+    else:
+        exit('ERROR: Wrong path or wrong file name included: %s' % loc)
+
+    log.info('moff Input file: %s  XIC_tol %s XIC_win %4.4f moff_rtWin_peak %4.4f ' % (file_name, tol, h_rt_w, s_w))
+    log.info('RAW file  :  %s' % (loc))
+    log.info('moff Input file: %s  XIC_tol %s XIC_win %4.4f moff_rtWin_peak %4.4f ', file_name, tol, h_rt_w, s_w)
+    log.info('Output_file in :  %s', outputname)
+    log.info('RAW file and its location :  %s', loc)
+    # read data from file
+    data_ms2 = pd.read_csv(file_name, sep="\t", header=0)
+    if check_columns_name(data_ms2.columns.tolist(), ast.literal_eval(config.get('moFF', 'col_must_have_x'))) == 1:
+        exit('ERROR minimal field requested are missing or wrong')
+
+    # fast debug ---
+    data_ms2 = data_ms2.head(100)
```

Comment:
this file should be deleted. I used to load only the first 100 feature for a fast debug